### PR TITLE
Ensure dev overlay uses basePath for requests

### LINKED ISSUE #14470: react-dev-overlay does not work correctly with `basePath`
# Bug report

## Describe the bug

react-dev-overlay does not work as expected when it is run with [`basePath`](https://github.com/vercel/next.js/issues/4998) because it sends a request to the` /__nextjs_original-stack-frame` and `/__nextjs_launch-editor` paths. however, when basePath is active, requests get 404 error because these paths are not actually available.

https://github.com/vercel/next.js/blob/7f6c6f031611057d44b253fbd53473471a84504f/packages/react-dev-overlay/src/middleware.ts#L83

https://github.com/vercel/next.js/blob/7f6c6f031611057d44b253fbd53473471a84504f/packages/react-dev-overlay/src/middleware.ts#L199

https://github.com/vercel/next.js/blob/7f6c6f031611057d44b253fbd53473471a84504f/packages/react-dev-overlay/src/internal/helpers/stack-frame.ts#L56

https://github.com/vercel/next.js/blob/86160a5190c50ea315c7ba91d77dfb51c42bc65f/packages/react-dev-overlay/src/internal/container/RuntimeError.tsx#L27

https://github.com/vercel/next.js/blob/86160a5190c50ea315c7ba91d77dfb51c42bc65f/packages/react-dev-overlay/src/internal/components/CodeFrame/CodeFrame.tsx#L49

## To Reproduce

`next.config.js`
```js
module.exports = {
  experimental: {
    basePath: '/blog',
  },
};
```

`pages/index.js`
```js
export default function Index() {
  return (
    <div>
      <h1>Hi!</h1>
      {foo}
    </div>
  );
}

```

image

image

## Expected behavior

image

## System information

- OS: macOS
- Browser chrome
- Version of Next.js: 9.4.4
- Version of Node.js: 14.3.0

## Additional context

I have resolved my problem temporarily by creating a custom server with the following method.

```js
const express = require('express');
const next = require('next').default;
const port = parseInt(process.env.PORT, 10) || 3000;

const nextServer = next({
  dir: '.',
  dev: process.env.NODE_ENV !== 'production',
});

const handler = nextServer.getRequestHandler();
const app = express();

nextServer.prepare().then(() => {
  app.use((req, res) => {
    if (
      req.originalUrl.startsWith('/__nextjs_original-stack-frame') ||
      req.originalUrl.startsWith('/__nextjs_launch-editor')
    ) {
      req.url = '/blog' + req.originalUrl;
    }
    return handler(req, res);
  });
  app.listen(port, (err) => {
    if (err) {
      throw err;
    }
    console.log(`> Ready on port ${port} `);
  });
});

```

### PR DESCRIPTION
**Pull request by ijjk**
This fixes the `basePath` not being used when making requests for the `react-dev-overlay`. We might want to pass down the `basePath` to the overlay instead of relying on the `process.env` injection although may be fine this way, will defer to @Timer for preferred way to pass this value to the dev-overlay

Closes: https://github.com/vercel/next.js/issues/14470

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| buildDuration | 11.7s | 11.8s | ⚠️ +128ms |
| nodeModulesSize | 67.1 MB | 67.1 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 1.996 | 1.998 | 0 |
| / avg req/sec | 1252.57 | 1251.12 | -1.45 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.178 | 1.162 | -0.02 |
| /error-in-render avg req/sec | 2121.81 | 2151.9 | ⚠️ +30.09 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.52 kB | 6.52 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| 19b7e98f51cc..62e5.js gzip | 10.6 kB | 10.6 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57 kB | 57 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.61 kB | 5.61 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| 19b7e98f51cc..dule.js gzip | 6.93 kB | 6.93 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.4 kB | 52.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| _buildManifest.js gzip | 267 B | 267 B | ✓ |
| _buildManife..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 539 B | 539 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| index.html gzip | 956 B | 956 B | ✓ |
| link.html gzip | 961 B | 961 B | ✓ |
| withRouter.html gzip | 948 B | 948 B | ✓ |
| Overall change | 2.87 kB | 2.87 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| buildDuration | 13.1s | 12.8s | -373ms |
| nodeModulesSize | 67.1 MB | 67.1 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.52 kB | 6.52 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| 19b7e98f51cc..62e5.js gzip | 10.6 kB | 10.6 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57 kB | 57 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.61 kB | 5.61 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| 19b7e98f51cc..dule.js gzip | 6.93 kB | 6.93 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.4 kB | 52.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| _buildManifest.js gzip | 267 B | 267 B | ✓ |
| _buildManife..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 539 B | 539 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| _error.js | 875 kB | 875 kB | ✓ |
| 404.html | 4.17 kB | 4.17 kB | ✓ |
| hooks.html | 3.79 kB | 3.79 kB | ✓ |
| index.js | 875 kB | 875 kB | ✓ |
| link.js | 914 kB | 914 kB | ✓ |
| routerDirect.js | 906 kB | 906 kB | ✓ |
| withRouter.js | 906 kB | 906 kB | ✓ |
| Overall change | 4.48 MB | 4.48 MB | ✓ |

</details>

</details>
Commit: b3cf533f4ab676c7dd1c550e66ddc55b44155a36

File: test/integration/basepath/test/index.test.js [hunk 1]
```diff
@@ -32,7 +34,16 @@ jest.setTimeout(1000 * 60 * 2)
...
 const runTests = (context, dev = false) => {
-  if (!dev) {
+  if (dev) {
+    it('should render error in dev overlay correctly', async () => {
+      const browser = await webdriver(context.appPort, '/docs/hello')
+      await browser.elementByCss('#trigger-error').click()
+      expect(await hasRedbox(browser)).toBe(true)
+
+      const errorSource = await getRedboxSource(browser)
+      expect(errorSource).toContain('oops heres an error')
```
Comment (Timer) [inline test/integration/basepath/test/index.test.js:44]: This test needs to capture the stack frame and sure it was resolved.

**Review by Timer — CHANGES_REQUESTED** [review]

File: test/integration/basepath/test/index.test.js (hunk 1 above)
Comment (ijjk) [reply test/integration/basepath/test/index.test.js]: Updated to check the full stack frame

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| buildDuration | 11.1s | 11.2s | ⚠️ +88ms |
| nodeModulesSize | 67.1 MB | 67.1 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 1.946 | 1.854 | -0.09 |
| / avg req/sec | 1284.91 | 1348.38 | ⚠️ +63.47 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.104 | 1.148 | ⚠️ +0.04 |
| /error-in-render avg req/sec | 2265.48 | 2177.17 | -88.31 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.52 kB | 6.52 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| 19b7e98f51cc..62e5.js gzip | 10.6 kB | 10.6 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57 kB | 57 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.61 kB | 5.61 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| 19b7e98f51cc..dule.js gzip | 6.93 kB | 6.93 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.4 kB | 52.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| _buildManifest.js gzip | 267 B | 267 B | ✓ |
| _buildManife..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 539 B | 539 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| index.html gzip | 956 B | 956 B | ✓ |
| link.html gzip | 961 B | 961 B | ✓ |
| withRouter.html gzip | 948 B | 948 B | ✓ |
| Overall change | 2.87 kB | 2.87 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| buildDuration | 12.1s | 11.7s | -382ms |
| nodeModulesSize | 67.1 MB | 67.1 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.52 kB | 6.52 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| 19b7e98f51cc..62e5.js gzip | 10.6 kB | 10.6 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57 kB | 57 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.61 kB | 5.61 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| 19b7e98f51cc..dule.js gzip | 6.93 kB | 6.93 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.4 kB | 52.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| _buildManifest.js gzip | 267 B | 267 B | ✓ |
| _buildManife..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 539 B | 539 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| _error.js | 875 kB | 875 kB | ✓ |
| 404.html | 4.17 kB | 4.17 kB | ✓ |
| hooks.html | 3.79 kB | 3.79 kB | ✓ |
| index.js | 875 kB | 875 kB | ✓ |
| link.js | 914 kB | 914 kB | ✓ |
| routerDirect.js | 906 kB | 906 kB | ✓ |
| withRouter.js | 906 kB | 906 kB | ✓ |
| Overall change | 4.48 MB | 4.48 MB | ✓ |

</details>

</details>
Commit: 20e16ec74cc11689d54794f110855ee06103b611

**Review by Timer — CHANGES_REQUESTED** [review]
Needs win32 test in branch
image

**Review by timneutkens — DISMISSED** [review]

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| buildDuration | 14.2s | 15.2s | ⚠️ +1.1s |
| nodeModulesSize | 67.1 MB | 67.1 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.465 | 2.558 | ⚠️ +0.09 |
| / avg req/sec | 1014.15 | 977.38 | -36.77 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.731 | 1.843 | ⚠️ +0.11 |
| /error-in-render avg req/sec | 1444.3 | 1356.24 | -88.06 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.49 kB | 6.49 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| 19b7e98f51cc..14a7.js gzip | 10.7 kB | 10.7 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57.1 kB | 57.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.57 kB | 5.57 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| 19b7e98f51cc..dule.js gzip | 7.07 kB | 7.07 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.5 kB | 52.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| _buildManifest.js gzip | 266 B | 266 B | ✓ |
| _buildManife..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 538 B | 538 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| index.html gzip | 955 B | 955 B | ✓ |
| link.html gzip | 962 B | 962 B | ✓ |
| withRouter.html gzip | 947 B | 947 B | ✓ |
| Overall change | 2.86 kB | 2.86 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| buildDuration | 15s | 15.4s | ⚠️ +446ms |
| nodeModulesSize | 67.1 MB | 67.1 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.49 kB | 6.49 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| 19b7e98f51cc..14a7.js gzip | 10.7 kB | 10.7 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57.1 kB | 57.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.57 kB | 5.57 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| 19b7e98f51cc..dule.js gzip | 7.07 kB | 7.07 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.5 kB | 52.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| _buildManifest.js gzip | 266 B | 266 B | ✓ |
| _buildManife..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 538 B | 538 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/basepath-dev-overlay | Change |
| - | - | - | - |
| _error.js | 875 kB | 875 kB | ✓ |
| 404.html | 4.17 kB | 4.17 kB | ✓ |
| hooks.html | 3.79 kB | 3.79 kB | ✓ |
| index.js | 875 kB | 875 kB | ✓ |
| link.js | 916 kB | 916 kB | ✓ |
| routerDirect.js | 908 kB | 908 kB | ✓ |
| withRouter.js | 908 kB | 908 kB | ✓ |
| Overall change | 4.49 MB | 4.49 MB | ✓ |

</details>

</details>
Commit: 843473ec38beb6d81b9b2fd8231798d7783846d7

**Review by Timer — APPROVED** [review]

**Pull request merged**